test/org/zaproxy/zap/extension/pscanrulesAlpha/XChromeLoggerDataInfoLeakScannerUnitTest.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2018 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+import java.io.IOException;
+
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.junit.Before;
+import org.junit.Test;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+public class XChromeLoggerDataInfoLeakScannerUnitTest extends PassiveScannerTest {
+
+	private static final String XCLD = "X-ChromeLogger-Data";
+	private static final String XCPD = "X-ChromePhp-Data";
+	private static final String XCLD_VALUE = "eyJ2ZXJzaW9uIjoiNC4wIiwiY29sdW"
+			+ "1ucyI6WyJsYWJlbCIsImxvZyIsImJhY2t0cmFjZSIsInR5cGUiXSwicm93cyI"
+			+ "6W1sicmVxdWVzdCIsIk1hdGNoZWQgcm91dGUgXCJhcHBfc2VjdXJpdHlfbG9n"
+			+ "aW5cIiAocGFyYW1ldGVyczogXCJfY29udHJvbGxlclwiOiBcIkJhY2tFbmRcX"
+			+ "EFwcEJ1bmRsZVxcQ29udHJvbGxlclxcU2VjdXJpdHlDb250cm9sbGVyOjpsb2"
+			+ "dpbkFjdGlvblwiLCBcIl9yb3V0ZVwiOiBcImFwcF9zZWN1cml0eV9sb2dpblw"
+			+ "iKSIsInVua25vd24iLCJpbmZvIl0sWyJzZWN1cml0eSIsIlBvcHVsYXRlZCBT"
+			+ "ZWN1cml0eUNvbnRleHQgd2l0aCBhbiBhbm9ueW1vdXMgVG9rZW4iLCJ1bmtub"
+			+"3duIiwiaW5mbyJdXSwicmVxdWVzdF91cmkiOiJcL2xvZ2luIn0=";
+	private HttpMessage msg;
+
+	@Before
+	public void before() throws URIException {
+		HttpRequestHeader requestHeader = new HttpRequestHeader();
+		requestHeader.setURI(new URI("http://example.com", false));
+
+		msg = new HttpMessage();
+		msg.setRequestHeader(requestHeader);
+	}
+
+	@Override
+	protected PluginPassiveScanner createScanner() {
+		return new XChromeLoggerDataInfoLeakScanner();
+	}
+
+	@Test
+	public void shouldNotRaiseAlertIfResponseHasNoRelevantHeader() {
+		// Given
+		msg.setResponseBody("Some text <h1>Some Title Element</h1>");
+		// When
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+		// Then
+		assertThat(alertsRaised.size(), equalTo(0));
+	}
+
+	@Test
+	public void shouldRaiseAlertIfResponseHasRelevantHeader() throws IOException {
```

Comment:
Could assert that the other info contains the decoded base64 value (same for following test).

Reply:
Will do.